client/witness.go

```diff
@@ -0,0 +1,53 @@
+package client
+
+import (
+	"github.com/docker/notary/client/changelist"
+	"github.com/docker/notary/tuf"
+	"path/filepath"
+)
+
+// Witness creates change objects to witness (i.e. resign) the given
+// roles on the next publish. One change is created per role
+func (r *NotaryRepository) Witness(roles ...string) ([]string, error) {
+	cl, err := changelist.NewFileChangelist(filepath.Join(r.tufRepoPath, "changelist"))
+	if err != nil {
+		return nil, err
+	}
+	defer cl.Close()
+
+	successful := make([]string, 0, len(roles))
+	for _, role := range roles {
+		// scope is role
+		c := changelist.NewTUFChange(
+			changelist.ActionUpdate,
+			role,
+			changelist.TypeWitness,
+			"",
+			nil,
+		)
+		err = cl.Add(c)
+		if err != nil {
+			break
+		}
+		successful = append(successful, role)
+	}
+	return successful, err
+}
+
+func witnessTargets(repo *tuf.Repo, invalid *tuf.Repo, role string) error {
+	if r, ok := repo.Targets[role]; ok {
+		// role is already valid, mark for re-signing/updating
+		r.Dirty = true
+		return nil
+	}
+	if invalid == nil {
+		// role isn't recognized, even as invalid
+		return ErrInvalidLocalRole{Role: role}
```

Comment:
Yes apologies, that one is just used for key rotation.  :|  Also, can we add tests for this case where we try to witness a role that doesn't exist?   Or even if it did exist once, but the user explicitly deleted it (i.e. it has been removed from the snapshot)?

Also would be great to test witnessing a valid role too.  :)